larsim/SimFilters/FilterSimPhotonLiteTime_module.cc

```diff
@@ -0,0 +1,170 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       FilterSimPhotonLiteTime
+// Module Type: filter
+// File:        FilterSimPhotonLiteTime_module.cc
+//
+// Author: Gray Putnam -- ported from FilterSimPhotonTime
+//
+// Module for filtering events based on the number of true photons
+// hitting optical detectors inside a time window. Uses the
+// sim::SimPhotonsLite data product as input (see FilterSimPhotonTime
+// for filtering using the sim::SimPhotons data product).
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Core/SharedFilter.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "fhiclcpp/ParameterSet.h"
+
+#include <memory>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "lardataobj/Simulation/SimPhotons.h"
+#include "lardataobj/Simulation/SimPhotons.h"
+
+namespace simfilter {
+  class FilterSimPhotonLiteTime;
+}
+
+class simfilter::FilterSimPhotonLiteTime : public art::SharedFilter {
+public:
+  explicit FilterSimPhotonLiteTime(fhicl::ParameterSet const& p,
+                               art::ProcessingFrame const&);
+
+  // Plugins should not be copied or assigned.
+  FilterSimPhotonLiteTime(FilterSimPhotonLiteTime const&) = delete;
+  FilterSimPhotonLiteTime(FilterSimPhotonLiteTime&&) = delete;
+  FilterSimPhotonLiteTime& operator=(FilterSimPhotonLiteTime const&) = delete;
+  FilterSimPhotonLiteTime& operator=(FilterSimPhotonLiteTime&&) = delete;
+
+private:
+  bool filter(art::Event& e, art::ProcessingFrame const&) override;
+
+  std::string const fSimPhotonsLiteCollectionLabel; //!< Label for the sim::SimPhotonsLite data product
+  std::vector<std::vector<int>> const fTimeWindows; //!< Time windows used for filtering. Units are the same as in the sim::SimPhotonsLite
+  int const fMinTotalPhotons; //!< Minimum number of photons inside a window to pass the filter
+  bool const fDebug; //!< Set to true to print (a lot of) debug information.
+  std::size_t const fN; //!< Number of time winows.
+  bool fUseReflectedPhotons; //!< Whether to include reflected photons in the filter.
+  std::string fReflectedLabel; //!< Label for the reflected photons -- "Reflected" by default.
+
+  void CheckTimeWindows() const;
+};
+
+simfilter::FilterSimPhotonLiteTime::FilterSimPhotonLiteTime(
+  fhicl::ParameterSet const& p,
+  art::ProcessingFrame const&)
+  : SharedFilter{p}
+  , fSimPhotonsLiteCollectionLabel(p.get<std::string>("SimPhotonsLiteCollectionLabel"))
+  , fTimeWindows(p.get<std::vector<std::vector<int>>>("TimeWindows"))
```

Comment:
Based on the below code, it looks like each vector element --needs-- to be length two.  You could naturally enforce this by using `std::vector<std::pair<float,float> >`

Reply:
Fixed.